src/test/resources/com/jpinpoint/perf/lang/java/ruleset/common/xml/UnconditionalCreatedLogArguments.xml

```diff
@@ -358,6 +358,67 @@ public class Issue468_PublicLog {
         }
     }
 }
+]]></code>
+    </test-code>
+
+    <test-code>
+        <description>Fix Request: UnconditionalCreatedLogArguments should recognize record getters. #569</description>
+        <expected-problems>1</expected-problems>
+        <expected-linenumbers>14</expected-linenumbers>
+        <code><![CDATA[
+import org.slf4j.Logger;
+class Issue569 {
+    private Logger log;
+
+    void bar1(MyRecord obj) {
+      String logObj = obj.topic();
+      log.debug("record topic: {}", logObj); // <- false positive
+    }
+
+    void bar2(WannabeRecord obj) {
+      String logObj = obj.topic();
+      log.debug("wannabe record topic: {}", logObj); // <- false positive
+      long logObj2 = obj.doStuff(5);
+      log.debug("wannabe record done stuff: {}", logObj2); // bad
+    }
+
+   void bar3() {
+      WannabeRecord obj = newWR();
+      String logObj = obj.topic();
+      log.debug("wannabe record topic: {}", logObj); // <- false positive
+    }
+}
+
+record MyRecord (
+    String topic,
+    Long documentId,
+    long offset
+) {}
+
+class WannabeRecord {
```

Comment:
does it also work with inner-WannabeRecord classes?

or usage of the WannabeRecord inside inner classes?

Reply:
Yes, I added a test for that to make sure